src/geop/services/src/main/scala/com/azavea/mmw/GeopServiceActor.scala

```diff
@@ -0,0 +1,69 @@
+package com.azavea.mmw
+
+import akka.actor.Actor
+import spray.routing._
+import spray.http._
+
+object UnattractiveButTemporary {
+  import geotrellis.raster.io.geotiff.reader._
+  import geotrellis.raster._
+  import geotrellis.raster.reproject._
+  import geotrellis.vector._
+  import geotrellis.vector.reproject._
+  import geotrellis.vector.io.json._
+  import geotrellis.spark.utils.SparkUtils
+  import geotrellis.spark._
+  import geotrellis.spark.tiling._
+  import geotrellis.raster.mosaic._
+  import geotrellis.proj4._
+
+  import org.apache.spark._
+  import org.apache.spark.{SparkConf, SparkContext}
+
+  import spray.json._
+  import spray.json.DefaultJsonProtocol._
+  import spray.httpx.unmarshalling._
+
+  import DataFiles._
+  import NonHistogram._
+
+  implicit val sc: SparkContext = SparkUtils.createSparkContext("Geoprocessing", new SparkConf())
+  val nlcd = DataFiles.delawareNLCD
+  val soil = DataFiles.delawareSoil
+
+  def jsonHistogram(str: String) : String = {
+    val polygon = str.parseJson.convertTo[Polygon]
+    val histogram: List[((Int, Int), Int)] = NonHistogram.gtHistogram(nlcd, soil, polygon).toList
+    histogram.toJson.prettyPrint
+  }
+}
+
+class GeopServiceActor extends Actor with GeopService {
+  def actorRefFactory = context
+  def receive = runRoute(myRoute)
+}
+
+trait GeopService extends HttpService {
+  import scala.util._
+
+  import MediaTypes._
+  import UnattractiveButTemporary._
+
+  val myRoute =
```

Comment:
`spark-jobserver` is very attractive, but there are a couple of things to keep in mind with it.

- Some requests do no to use spark-context at all and will instead use `tileReader` to work on one tile at a time.
- Currently requests are expected to to terminate in request-response cycle, so job monitoring is mute.
- We have not use it yet and its not  clear what the limitations for job parameters are.

I think one we have a situation where the request can not be complete in request-response cycle it is a really good idea to start playing the the `spark-jobserver`. Particular interesting is if we can use the named/cached RDDs as a caching layer for calculations that cover a large area. This would probably require having a way to fetch those tiles in some way other than TMS service and placing them on a Leaflet.